### DISCUSSION TITLE
Performance Improvement

### DISCUSSION THREAD

**Discussion by bangcorrupt** [bangcorrupt/hacktribe#242] [Announcements]
If you have experienced performance issues with Hacktribe, try the latest commit [(6f48288)](https://github.com/bangcorrupt/hacktribe/commit/6f4828832eb8b1cc04f93965c143fc4c45cc2ab5) to see if they are fixed.  Hacktribe should no longer hang when changing Oscillator and FX under heavy DSP load.  There are still some timing issues when scrolling through oscillators and FX, but these are also present in the factory firmware.

**Comment by tripsylon** [comment]
Helo, first of all tank You for the great work, Its a rebirth for the e2, es2 community, once again thank You. But I have a question, do You think that there could be something done with the system lagging when you switch samples during playback?

**Reply by bangcorrupt** [reply]
With the latest update performance is similar to the factory firmware, so no guarantee I'll be able to improve it.  I have noticed that it only lags while the parameter display pop up is showing, so maybe a conflict with timer priority preventing the CPU sending updates to the DSP in time.  It lags for all parameter changes that show a pop up, but it's worst for sample changing.

**Reply by tripsylon** [reply]
I thought so... Maybe a solution would be to write the samples on to the ram? Im thinking about that because on the e2 switching between sounds doesent make problems.
